src/web3id.rs

```diff
@@ -181,3 +180,172 @@ pub async fn get_public_data(
+        }
+    }
+
+    #[derive(concordium_std::Serial, concordium_std::Deserial, Eq, PartialEq)]
+    /// The credential stored in the credential storage contract.
+    pub struct VersionedEncryptedCredential {
+        /// Metadata associated with the credential.
+        version:              u16,
+        /// The encrypted credential.
+        encrypted_credential: Vec<u8>,
+    }
+
+    #[derive(Debug, thiserror::Error)]
+    pub enum DecryptError {
+        #[error("Unsupported version: {0}")]
+        UnsupportedVersion(u16),
+        #[error("Unable to parse response: {0}")]
+        Parse(#[from] concordium_std::ParseError),
+        #[error("Unable to decrypt: {0}")]
+        DecryptFailure(#[from] aes_gcm::Error),
+        #[error("Unable decode decrypted credential secrets.")]
+        DecodeFailure,
+    }
+
+    impl VersionedEncryptedCredential {
+        /// Decrypt the [`ViewResponse`] with the provided AES256 secret key.
+        /// The public key `pk` is used to check authenticity.
+        pub fn decrypt(
+            &self,
+            pk: CredentialHolderId,
+            key: [u8; 32],
+        ) -> Result<CredentialSecrets, DecryptError> {
+            if self.version != 0 {
+                return Err(DecryptError::UnsupportedVersion(self.version));
+            }
+            let encrypted = concordium_std::from_bytes::<EncryptedCredentialSecrets>(
+                &self.encrypted_credential,
+            )?;
+            let cipher = Aes256Gcm::new(&key.into());
+            let payload = aes_gcm::aead::Payload {
+                msg: &encrypted.ciphertext,
+                aad: pk.public_key.as_bytes(),
+            };
+            let decrypted = cipher.decrypt(&encrypted.nonce.into(), payload)?;
+            let Ok(cs) = common::from_bytes(&mut std::io::Cursor::new(decrypted)) else {
+            return Err(DecryptError::DecodeFailure);
+        };
+            Ok(cs)
+        }
+    }
+
+    /// The parameter type for the contract function `store`.
+    #[derive(concordium_std::Serialize, serde::Serialize, serde::Deserialize, Debug, Clone)]
+    pub struct StoreParam {
+        /// Public key that created the above signature.
+        pub public_key: CredentialHolderId,
+        /// Signature from the holder.
+        #[serde(with = "crate::internal::byte_array_hex")]
+        pub signature:  [u8; ed25519_dalek::SIGNATURE_LENGTH],
+        // The signed data.
+        pub data:       DataToSign,
+    }
+
+    /// The parameter type for the contract function `serializationHelper`.
+    #[derive(concordium_std::Serialize, serde::Serialize, serde::Deserialize, Clone, Debug)]
+    pub struct DataToSign {
+        /// A timestamp to make signatures expire.
+        pub timestamp:            Timestamp,
+        /// The contract_address that the signature is intended for.
+        pub contract_address:     ContractAddress,
+        /// Metadata associated with the credential.
+        pub version:              u16,
```

Comment:
This field is called `metadata` in the credential storage contract, but it's called "version" here. Maybe it should be metadata everywhere for consistency, as the documentation above says.

Reply:
Version is what it is, so I want to keep it like this in the SDK API.

Maybe you want to make a PR changing the name in the contract?

Reply:
I have updated the documentation.